analyzers/src/SonarAnalyzer.Common/Json/JsonNode.cs

```diff
@@ -98,6 +98,9 @@ public void Add(JsonNode value)
         public bool ContainsKey(string key) =>
             Kind == Kind.Object ? map.ContainsKey(key) : throw InvalidKind();
 
+        public bool TryGetPropertyValue(string key, out JsonNode node) =>
+            Kind == Kind.Object ? map.TryGetValue(key, out node) : throw InvalidKind();
```

Comment:
The json property type can be value, object or list. I don't see any reason to limit the kind to `object`. Maybe it is better to name the method `TryGetPropertyNode` to make this more explicit. Further, I wouldn't expect a `Try` method to throw at all.
```suggestion
            map.TryGetValue(key, out node);
```